src/transformers/pipelines.py

```diff
@@ -516,7 +516,7 @@ def __init__(
         if framework is None:
             framework = get_framework(model)
 
-        self.task = task
+        self.task = task or self.task
```

Comment:
This is not defined for `Pipeline`  and most others, so we probably should change that to 

`getattr(self, 'task', "")`